File: easy/slang_flavor.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

var (
	phrases = []string{", yeah!", ", this is crazy, I tell ya.",
		", can U believe this?", ", eh?", ", aw yea.",
		", yo.", "? No way!", ". Awesome!"}
)

func slang(q string, c int, l bool) (string, int, bool) {
	if q == "." || q == "!" || q == "?" {
		if l {
			return phrases[c], (c + 1) % len(phrases), false
		}
		return q, c, true
	}
	return q, c, l
}

func main() {
	var (
		c int
		l bool
		t string = "\n"
	)
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	scanner := bufio.NewScanner(data)
	scanner.Split(bufio.ScanBytes)
	for scanner.Scan() {
		t, c, l = slang(scanner.Text(), c, l)
		fmt.Print(t)
	}
	if t != "\n" {
		fmt.Println()
	}
}

File: easy/slang_flavor.hs
import System.Environment (getArgs)

phrases :: [String]
phrases = [", yeah!", ", this is crazy, I tell ya.",
    ", can U believe this?", ", eh?", ", aw yea.",
    ", yo.", "? No way!", ". Awesome!"]

slang           :: Int -> Bool -> String -> String
slang _ _ []     = ""
slang i l (x:xs) | elem x ".!?" && l = phrases!!i ++ slang (mod (i+1) 8) False xs
                 | elem x ".!?"      = x : slang i True xs
                 | otherwise         = x : slang i l xs

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr $ slang 0 False (if last input == '\n' then input else input ++ "\n")
